tests/by-util/test_env.rs

```diff
@@ -882,6 +882,33 @@ fn test_env_arg_ignore_signal_empty() {
         .no_stderr()
         .stdout_contains("hello");
 }
+
+#[test]
+fn disallow_equals_sign_on_short_options() {
```

Comment:
I would rename the function to highlight that it only affects `-u` (and not all short options).